src/common/line_entry.rs

```diff
@@ -72,6 +72,54 @@ impl LineEntry {
         }
         comment::parse(self.raw_string.as_str())
     }
+
+    #[allow(dead_code)]
+    pub fn get_substitution_keys(&self) -> Vec<&str> {
+        let mut keys = Vec::new();
+
+        let mut value = match self.get_value().map(str::trim) {
+            Some(value) if !value.starts_with('\'') => value,
+            _ => return keys,
+        };
+
+        let is_escaped =
+            |prefix: &str| prefix.chars().rev().take_while(|ch| *ch == '\\').count() % 2 == 1;
+
+        if value.starts_with('\"') {
+            if value.ends_with('\"') && !is_escaped(&value[..value.len() - 1]) {
+                value = &value[1..value.len() - 1]
+            } else {
+                return keys;
+            }
+        }
+
+        while let Some(index) = value.find('$') {
+            let prefix = &value[..index];
+            let raw_key = &value[index + 1..];
+
+            if is_escaped(prefix) {
+                value = &raw_key;
+            } else {
+                let (key, rest) = raw_key
+                    .strip_prefix('{')
+                    .and_then(|raw_key| raw_key.find('}').map(|i| raw_key.split_at(i)))
+                    .or_else(|| {
+                        raw_key
+                            .find(|c: char| !c.is_ascii_alphanumeric() && c != '_')
+                            .map(|i| raw_key.split_at(i))
+                    })
+                    .unwrap_or((raw_key, ""));
+                if !key.is_empty() {
+                    keys.push(key);
+                } else {
+                    return keys;
+                }
```

Comment:
Yeah, that's good point!
```suggestion
                if key.is_empty() {
                    return keys;
                }
                
                keys.push(key);

```

Reply:
Done